FrenchFilter/sections/general_extensions.txt

```diff
@@ -37,6 +37,14 @@ actu17.fr#%#//scriptlet("abort-on-property-write", "td_ad_background_click_link"
 jacquieetmicheltv.net#%#var _st = window.setTimeout; window.setTimeout = function(a, b) { if(!/target_url/.test(a)){ _st(a,b);}};
 ! https://github.com/AdguardTeam/AdguardFilters/issues/83543
 canalplus.com#%#//scriptlet('set-constant', '__data.application.settings.featPlayerAds', 'false')
+! https://github.com/AdguardTeam/AdguardFilters/issues/100525
+prod-player.tf1.fr#%#(function(){let callCounter=0;const nativeJSONParse=JSON.parse;const handler={apply(){callCounter++;const obj=Reflect.apply(...arguments);if(callCounter<=10){if(obj.hasOwnProperty('env')&&obj.env.hasOwnProperty('origin')){if(!obj.hasOwnProperty('ads')){obj.ads={};}obj.ads.enable=false;obj.ads._prerolls=false;obj.ads._midrolls=false;}}else{JSON.parse=nativeJSONParse;}return obj;}};JSON.parse=new Proxy(JSON.parse,handler);})();

```

Comment:
Please add a comment with this link as well:
https://gist.github.com/ameshkov/265f353a816110e8b9d5fc37ad66c31a

Also, could you please explain the logic?

I am just trying to figure out why `json-prune` is not an option here

Reply:
Hi! OK, added!

For the logic, see the section `Without going into too much detail …`.

Basically, I don't want to delete the ads one by one using `json-prune` but to disable the ads entirely, on this catch-up TV web app. To do this, you must modify the configuration (web app init stuff). In the case of TF1, even before the appearance of this one, a very light JSON data appearing almost immediately and containing only `[ROOT].env.origin` (see screenshots present in the section cited above) is found to be merged on top of the config. Perfect for grafting the desired parameters inside! In addition, this solution allows that after 10 calls to `JSON.parse`, the native code is restored. Top for performance (important on this kind of web app).

And concerning the fact of waiting for the main config JSON to pass and to do a `json-prune` action when found on the desired keys, already tried — in fact it was among my first tests after a check using `json-prune` without parameters in order to analyze the situation — (and tried again now just to be sure): does not work unlike 6play / RTLplay where I remember it worked. TF1 makes sure that these values are _both_ present and boolean, unfortunately. So I have to inject the desired key-value pairs.

Reply:
Got it.

So to solve this in a declarative manner we need to provide a scriptlet that would override properties and not just remove them? Something like `json-set-constant`?

Reply:
Exactly. Or, if too risky whatever value people define here (including simple booleans, etc.) to be allowed in third-party lists, which I can understand, just a single `agtrusted-json-add`. For all of this, see the section `Convertable later into a scriptlet …` (which details exactly all this).

Reply:
Hi! You have done a lot there, thanks for your contribution. We also want to see new pull requests from you in the future :)

Reply:
I admit that I have never slept so little these days haha!